Makefile

```diff
@@ -458,6 +459,10 @@ endif
 
 rust-tests:
 	$(SHOW)cd $(REDISEARCH_RS_DIR) && $(RUST_TEST_RUNNER) test $(RUST_TEST_OPTIONS) $(TEST_NAME)
+ifeq ($(RUN_MIRI),1)
```

Comment:
this is the main comment from me on this PR:
from the PR description 
```
You need to opt into running tests through miri, with a dedicated RUN_MIRI option.
In CI it's enabled exclusively for the sanitize check (PR event flow + merge-queue event flow).
```
can you explain why this is not `on` by default when SAN is used so we can avoid additional flag to the makefile?
we can avoid the additional flags in the CI as well

Reply:
I didn't reuse `SAN` because Rust supports sanitizers (see [here](https://doc.rust-lang.org/stable/unstable-book/compiler-flags/sanitizer.html) for details), and we plan to honor that flag in the future.
Running with sanitizers in Rust is a much heavier/cumbersome operation (since it requires the `build-std` unstable flag) so it's convenient to be able to run `miri` without having to enable sanitizers.